src/main/scala/org/ergoplatform/nodeView/wallet/WalletScanLogic.scala

```diff
@@ -67,7 +69,10 @@ object WalletScanLogic extends ScorexLogging {
     val context = new ErgoContext(stateContext, transactionContext, inputContext,
       LaunchParameters.maxBlockCost, CostTable.interpreterInitCost)
 
-    proverOpt.flatMap(_.prove(box.ergoTree, context, testingTx.messageToSign).toOption).isDefined
+    proverOpt.flatMap { prover =>
+      prover.IR.resetContext()
```

Comment:
I missed this call of prove. Yes, this resetContext is necessary because the prover is part of actor state.